lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
Ah ok, I see what you mean, then the interface is the same.
However, what is the benefit of doing this instead of raising the original exception with extended string (to also avoid the additional overhead of catch-rethrow)?